softicar/platform
# PLAT-647 Create EmfTokenMatrixParser

- Created `EmfTokenMatrixParser` to derive `IEmfTableRow` instances from a `List` of `List`s of `String` based tokens, and a specified target table.
- Created token-to-object converters for all 14 relevant DB table field types.
  - Those are: `BigDecimal`, `Boolean`, `ByteArray`, `Day`, `DayTime`, `Double`, `Enum`, `Float`, `Foreign`, `ID`, `Integer`, `Long`, `String`, `Time`
  - Available converters are registered in `EmfTokenConverters`.
- Basic validation of `String` based tokens is performed.
  - Mode of operation is "fail-fast".
  - Encountered error messages are i18n'ed.
  - Error messages are crafted to be as helpful as possible, and to quickly point to the offending token.
  - Error message wording aims towards a system administrator, or even trained-end-user, audience.

Test Plan:
- check API of `EmfTokenMatrixParser`
- see `EmfTokenRowParserTest` and `EmfTokenMatrixParserExceptionBuilderTest`
